ard_controller/ard_controller.ino

```diff
@@ -245,20 +287,89 @@ void rx_processor() {
   }
 }
 
+// reads current position estimate from odrive and stores in global array
+void update_cur_pos() {
+  odrv_leftleg_ser.write("r axis0.encoder.pos_estimate\n"); // left hip
+  cur_joint_pos[0] = constrain(fmod(odrv_leftleg.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_leftleg_ser.write("r axis1.encoder.pos_estimate\n"); // left knee
+  cur_joint_pos[1] = constrain(fmod(odrv_leftleg.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_rightleg_ser.write("r axis0.encoder.pos_estimate\n"); // right hip
+  cur_joint_pos[2] = constrain(fmod(odrv_rightleg.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_rightleg_ser.write("r axis1.encoder.pos_estimate\n"); // right knee
+  cur_joint_pos[3] = constrain(fmod(odrv_rightleg.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+
+  odrv_leftarm_ser.write("r axis0.encoder.pos_estimate\n"); // left shoulder
+  cur_joint_pos[4] = constrain(fmod(odrv_leftarm.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_leftarm_ser.write("r axis1.encoder.pos_estimate\n"); // left elbow
+  cur_joint_pos[5] = constrain(fmod(odrv_leftarm.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_rightarm_ser.write("r axis0.encoder.pos_estimate\n"); // right shoulder
+  cur_joint_pos[6] = constrain(fmod(odrv_rightarm.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_rightarm_ser.write("r axis1.encoder.pos_estimate\n"); // right elbow
+  cur_joint_pos[7] = constrain(fmod(odrv_rightarm.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+}
+
+// test if the current position is at the goal position
+bool eval_at_goal() {
+  int l_hip = joint_angle_goal.left_hip;
+  int l_knee = joint_angle_goal.left_knee;
+  int r_hip = joint_angle_goal.right_hip;
+  int r_knee = joint_angle_goal.right_knee;
+  int l_sh = joint_angle_goal.left_shoulder;
+  int l_elb = joint_angle_goal.left_elbow;
+  int r_sh = joint_angle_goal.right_shoulder;
+  int r_elb = joint_angle_goal.right_elbow;
+
+  if (!(cur_joint_pos[0] <= l_hip + DEADBAND && cur_joint_pos[0] >= l_hip - DEADBAND)) {
```

Comment:
Your Deadband value is 3. Does that mean we are allowing the robot to be off by an angle of 3 degrees? Is that a little too much?

Reply:
This would be 3 degrees at the motor. Divided by 9, it would potentially be off by 1/3 degree. The deadband can be adjusted to whatever you feel necessary.